### PR TITLE
Feature/adn 842

### PR DESCRIPTION
**Pull request by JuliaGalabut**

### CODE REVIEW

File: oap-ws-sso-api/pom.xml [hunk 1]
```diff
@@ -55,6 +55,11 @@
             <artifactId>guava</artifactId>
             <version>${oap.deps.guava.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.auth0</groupId>
+            <artifactId>java-jwt</artifactId>
+            <version>3.19.2</version>
```
Comment (alebastrov) [inline oap-ws-sso-api/pom.xml:61]: move to properties  please
oap.deps.jwt.version

File: oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/AbstractAuthTokenProvider.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        this.issuer = issuer;
+    }
+
+    @Override
+    public boolean verifyToken( String token ) {
+        try {
+            final DecodedJWT decodedJWT = decodeJWT( token );
+            return decodedJWT != null;
+        } catch( JWTVerificationException e ) {
+            log.trace( "Token not valid:", e );
```
Comment (alebastrov) [inline oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/AbstractAuthTokenProvider.java:52]: maybe warn to have it visible?

File: oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    public JWTSecurityInterceptor( AbstractAuthTokenProvider tokenProvider ) {
+        this.tokenProvider = tokenProvider;
+    }
+
+    @Override
+    public Optional<Response> before( InvocationContext context ) {
+        final String authorization = context.exchange.getRequestHeader( AUTHENTICATION_KEY );
+        if( authorization == null || authorization.isEmpty() ) {
+            log.trace( "Not authenticated. No token in header {}", context.exchange );
```
Comment (alebastrov) [inline oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java:54]: return Optional.empty();
or
return Optional.of( new Response( FORBIDDEN, "Token is missing" )
just here?

File: oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        if( !tokenProvider.verifyToken( authorization ) ) {
+            log.trace( "Not authenticated." );
```
Comment (alebastrov) [inline oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java:57]: return Optional.empty();
or
return Optional.of( new Response( FORBIDDEN, "Invalid token" )
just here?

**Review by alebastrov — CHANGES_REQUESTED** [review]

File: oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java (hunk 3 above)
Comment (JuliaGalabut) [reply oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java:54]: it seems, we have methods which do not require authorization.

File: oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java (hunk 4 above)
Comment (JuliaGalabut) [reply oap-ws-sso-api/src/main/java/oap/ws/sso/interceptor/JWTSecurityInterceptor.java:57]: added

File: oap-ws-sso-api/pom.xml (hunk 1 above)
Comment (JuliaGalabut) [reply oap-ws-sso-api/pom.xml]: agreee

**Pull request merged**